src/Akka.Persistence.MongoDb/Journal/MongoDbJournal.cs

```diff
@@ -203,16 +199,30 @@ await _journalCollection.Value
             return maxOrderingId;
         }
 
+        /// <summary>
+        /// Asynchronously reads a highest sequence number of the event stream related with provided <paramref name="persistenceId"/>.
+        /// </summary>
+        /// <param name="persistenceId">TBD</param>
+        /// <param name="fromSequenceNr">TBD</param>
+        /// <returns>long</returns>
         public override async Task<long> ReadHighestSequenceNrAsync(string persistenceId, long fromSequenceNr)
         {
-            NotifyNewPersistenceIdAdded(persistenceId);
 
             var builder = Builders<MetadataEntry>.Filter;
             var filter = builder.Eq(x => x.PersistenceId, persistenceId);
 
-            var highestSequenceNr = await _metadataCollection.Value.Find(filter).Project(x => x.SequenceNr).FirstOrDefaultAsync();
+            //Following the SqlJournal implementation
+            //I have tried MongoDb lookup query and that caused some deadlocks in some tests!
+            
+            var metadataHighestSequenceNr = await _metadataCollection.Value.Find(filter).Project(x => x.SequenceNr).FirstOrDefaultAsync();
+
+            //var journalHighestSequenceNr = await _journalCollection.Value.Find(Builders<JournalEntry>.Filter.Eq(x => x.PersistenceId, persistenceId)).Project(x => x.SequenceNr).FirstOrDefaultAsync();
```

Comment:
So this is a breaking change, to some extent - I think we only added metadata entry back in 2016, so anyone who:

1. Has been using Akka.Persistence.MongoDb since prior to Akka.NET 1.1.1;
2. Has persistent entities that have not been updated since then; and
3. Needs to recover those entities at some point in the future.

This change will break the code for those users. I think the population of those users is pretty close to zero, so it may not be a big deal - and the upside is the number of network round-trips used during the initial recover is cut from two to one for the "determining sequence number" stage.

Reply:
cc @jackowild in case he has some thoughts on this